plugins/net_plugin/net_plugin.cpp

```diff
@@ -2987,10 +2978,10 @@ namespace eosio {
             auto c_time = last_handshake_sent.time;
             g_conn.unlock();
             peer_dlog( this, "checking for duplicate" );
-            std::shared_lock<std::shared_mutex> g_cnts( my_impl->connections_mtx );
-            for(const auto& check : my_impl->connections) {
+            bool dup = false;
+            my_impl->connections.for_each_connection([&](const auto& check) {
```

Comment:
Not sure that a method named `any` is any better than just `for_each_connection`. I Can certainly make an external lambda to avoid the bool dup. I don't see how we can remove the `if( !f( c ) ) return;` as it is used to short circuit out of the for loop.

Reply:
I am suggesting *adding* the `any` in addition to the `for_each_connection`. If using the `any` in that one location, the short-circuiting of the `for_each` will not be needed anymore.

Reply:
Oh, I think I misunderstood. See if what I have now matches what you are asking for.